_deploy/r/gnoswap/gnft/gnft.gno

```diff
@@ -148,75 +197,121 @@ func Render(path string) string {
 	}
 }
 
-// Util
-func assertIsAdmin(address std.Address) {
-	if address != admin {
-		panic(addDetailToError(
-			errNoPermission,
-			ufmt.Sprintf("gnft.gno__assertIsAdmin() || only admin(%s) can call this function, called from %s", admin.String(), address.String()),
-		))
-	}
-}
+// SetTokenURI sets the metadata URI for a specific token using random generation svg image
+// Parameters:
+//   - tid: The token ID to set the URI for
+//
+// Returns:
+//   - string: The token URI
+//   - error: Returns an error if the URI setting fails, nil otherwise
+func SetTokenURI(tid grc721.TokenID) (string, error) {
+	common.IsHalted()
 
-func Exists(tid grc721.TokenID) bool {
-	_, err := gnft.OwnerOf(tid)
+	tokenURI := genImageURI(generateRandInstnace())
+	ok, err := gnft.SetTokenURI(tid, grc721.TokenURI(tokenURI))
 	if err != nil {
-		return false
+		return "", err
 	}
 
-	return true
+	prevAddr, prevPkgPath := getPrevAsString()
+	std.Emit(
+		"SetTokenURI",
+		"prevAddr", prevAddr,
+		"prevRealm", prevPkgPath,
+		"lpTokenId", string(tid),
+		"internal_tokenURI", tokenURI,
+	)
+
+	return tokenURI, nil
 }
 
-func SetTokenURI(tid grc721.TokenID) {
+// SetTokenURILast sets metadata URI for last minted nft token
+// Returns:
+//   - error: Returns an error if the URI setting fails, nil otherwise
+func SetTokenURILast() error {
```

Comment:
When is it being used in GnoSwap? what's the difference between this and `SetTokenURI`?

Reply:
`SetTokenURI(tokenId)` function takes paramter
- which `tokenId` you want to set uri for

Reply:
On the other had, this `SetTokenURILast()` does not takes any token Id.

This function is used on gnoswap interface.
- To show gnoswap nft svg image(with bit of random color), token URI must be set.
- We do this, by making transaction with multi-msg using adena.
```
msg01 : mint new position
msg02 : set token uri last 
```

Unfortunately, in multi-msg transaction, each msg can not get return value from previous msg.
In other words, we can't use `SetTokenURI(tokenId)` because it require tokenID.
Therefore we use `SetTokenURILast()`, and this function does get token id automatically for you then set uri.